model/github_app_auth.go

```diff
@@ -0,0 +1,133 @@
+package model
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/githubapp"
+	"github.com/evergreen-ci/evergreen/util"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+// TODO (DEVPROD-11883): move all of this logic into the githubapp package once
+// all project GitHub apps are using Parameter Store and the rollout is stable.
+// The functions are only here temporarily to avoid a dependency cycle between
+// the githubapp and model packages due to the project ref feature flag.
+
+// githubAppCheckAndRunParameterStoreOp checks if the project corresponding to
+// the GitHub app auth has Parameter Store enabled, and if so, runs the given
+// Parameter Store operation.
+func githubAppCheckAndRunParameterStoreOp(ctx context.Context, appAuth *githubapp.GithubAppAuth, op func(ref *ProjectRef, isRepoRef bool), opName string) {
+	ref, isRepoRef, err := findProjectRef(appAuth.Id)
+	grip.Error(message.WrapError(err, message.Fields{
+		"message":    "could not check if Parameter Store is enabled for project; assuming it's disabled and will not use Parameter Store",
+		"op":         opName,
+		"project_id": appAuth.Id,
+		"epic":       "DEVPROD-5552",
+	}))
+	isPSEnabled, err := isParameterStoreEnabledForProject(ctx, ref)
+	grip.Error(message.WrapError(err, message.Fields{
+		"message":    "could not check if Parameter Store is enabled for project; assuming it's disabled and will not use Parameter Store",
+		"op":         opName,
+		"project_id": appAuth.Id,
+		"epic":       "DEVPROD-5552",
+	}))
+	if !isPSEnabled {
+		return
+	}
+
+	op(ref, isRepoRef)
+}
+
+// GitHubAppAuthUpsert upserts the GitHub app auth into the database and to
+// Parameter Store if enabled.
+func GitHubAppAuthUpsert(appAuth *githubapp.GithubAppAuth) error {
```

Comment:
nit: can unexport

Reply:
Done. I'm not sure it matter really though, since it's temporary code.